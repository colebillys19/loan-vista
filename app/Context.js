import React, { createContext, useState } from 'react';
import T from 'prop-types';

const initialState = { mainError: false, mainLoading: true };

const Context = createContext([{}, () => {}]);

export const ContextProvider = ({ children }) => {
  const [state, setState] = useState(initialState);

  return (
    <Context.Provider value={[state, setState]}>{children}</Context.Provider>
  );
};

ContextProvider.propTypes = { children: T.node.isRequired };

export default Context;
