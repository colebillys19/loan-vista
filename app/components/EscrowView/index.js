/**
 * EscrowView
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import EscrowEscrow from 'components/EscrowEscrow';
import EscrowHomeowners from 'components/EscrowHomeowners';
import EscrowMortgage from 'components/EscrowMortgage';
import EscrowTaxes from 'components/EscrowTaxes';
import ConditionalRender from 'components/_shared/ConditionalRender';
import ErrorModal from 'components/_shared/ErrorModal';
import TabContainer from 'components/_shared/TabContainer';

const EscrowView = ({
  error,
  escrowEscrowData,
  escrowHomeownersData,
  escrowMortgageData,
  escrowTaxesData,
}) => (
  <TabContainer aria-labelledby="escrow-tab" id="escrow-view">
    <EscrowEscrow data={escrowEscrowData} />
    <EscrowTaxes data={escrowTaxesData} />
    <EscrowHomeowners data={escrowHomeownersData} />
    <EscrowMortgage data={escrowMortgageData} />
    <ConditionalRender
      Component={ErrorModal}
      propsToPassDown={{ error }}
      shouldRender={!!error}
    />
  </TabContainer>
);

EscrowView.propTypes = {
  error: T.oneOfType([T.bool, T.string]).isRequired,
  escrowEscrowData: T.object.isRequired,
  escrowHomeownersData: T.array.isRequired,
  escrowMortgageData: T.array.isRequired,
  escrowTaxesData: T.array.isRequired,
};

export default EscrowView;
