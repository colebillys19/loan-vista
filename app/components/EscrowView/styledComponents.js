import styled from 'styled-components';

export const EscrowViewWrapper = styled.div``;
