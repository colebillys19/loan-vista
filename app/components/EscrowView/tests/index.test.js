import React from 'react';
import { render } from 'react-testing-library';

import { ContextProvider } from 'Context';

import EscrowView from '../index';

import {
  MOCK_ESCROW_DATA as escrowEscrowData,
  MOCK_HOMEOWNERS_DATA as escrowHomeownersData,
  MOCK_MORTGAGE_DATA as escrowMortgageData,
  MOCK_TAXES_DATA as escrowTaxesData,
} from './mockData';

const mockProps = {
  error: false,
  escrowEscrowData,
  escrowHomeownersData,
  escrowMortgageData,
  escrowTaxesData,
};

const Component = (
  <ContextProvider>
    <EscrowView {...mockProps} />
  </ContextProvider>
);

describe('<EscrowView />', () => {
  it('Expect not to log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(Component);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });
});
