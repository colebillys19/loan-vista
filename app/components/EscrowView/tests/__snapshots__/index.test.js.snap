// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EscrowView /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="escrow-tab"
  class="TabContainer-sc-1edk8ho-0 kmLNPJ"
  id="escrow-view"
>
  <section
    class="styledComponents__CardContainer-sc-1wpt79b-0 dABPYv"
  >
    <div
      class="styledComponents__TopStripe-sc-1wpt79b-4 klOucO"
      color="rgb(80, 103, 55)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1wpt79b-2 fLRQBK"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 1.8rem; width: 1.8rem;"
        viewBox="0 0 64 64"
      >
        <circle
          cx="32"
          cy="31.964"
          fill="rgb(255, 191, 0)"
          r="21"
        />
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1wpt79b-1 ikQNuj"
      >
        Escrow
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1wpt79b-3 eVQUHf"
      />
    </div>
    <div
      class="styledComponents__ListsContainer-vwc47z-1 gQnTSI"
    >
      <div
        class="styledComponents__StyledTwoColumnList-vwc47z-2 jesBqZ"
      >
        <div
          class="styledComponents__RowContainer-m3am64-0 bLaNeN"
        >
          <div
            class="styledComponents__RowStripe-m3am64-1 bqKTOk"
          />
        </div>
        <div
          class="styledComponents__RowContainer-m3am64-0 bLaNeN"
        >
          <div
            class="styledComponents__RowStripe-m3am64-1 bqKTOk"
          />
        </div>
        <div
          class="styledComponents__RowContainer-m3am64-0 bLaNeN"
        >
          <div
            class="styledComponents__RowStripe-m3am64-1 bqKTOk"
          />
        </div>
      </div>
      <div
        class="styledComponents__StyledTwoColumnList-vwc47z-2 jesBqZ"
      >
        <div
          class="styledComponents__RowContainer-m3am64-0 bLaNeN"
        >
          <div
            class="styledComponents__RowStripe-m3am64-1 bqKTOk"
          />
        </div>
        <div
          class="styledComponents__RowContainer-m3am64-0 bLaNeN"
        >
          <div
            class="styledComponents__RowStripe-m3am64-1 bqKTOk"
          />
        </div>
        <div
          class="styledComponents__RowContainer-m3am64-0 bLaNeN"
        >
          <div
            class="styledComponents__RowStripe-m3am64-1 bqKTOk"
          />
        </div>
      </div>
    </div>
    <div
      class="styledComponents__ChartsSection-vwc47z-0 hKeGca"
    >
      <div
        class="styledComponents__BlockContainer-sc-1b51i62-0 fYurXy"
      >
        <h3
          class="styledComponents__Heading-sc-1b51i62-2 frvBdD"
        >
          Current Escrow Breakdown
        </h3>
        <div
          class="styledComponents__ChartContainer-sc-1b51i62-1 keWEnA"
        >
          <div
            class="styledComponents__SkeletonContainer-sc-1b51i62-4 jVKzvb"
          >
            <svg
              height="0"
              style="position: absolute;"
              width="0"
            >
              <defs>
                <clippath
                  id="donutClipPath"
                >
                  <path
                    d="M100,0 C155.228475,0 200,44.771525 200,100 C200,155.228475 155.228475,200 100,200 C44.771525,200 0,155.228475 0,100 C0,44.771525 44.771525,0 100,0 Z M100,58 C76.8040405,58 58,76.8040405 58,100 C58,123.195959 76.8040405,142 100,142 C123.195959,142 142,123.195959 142,100 C142,76.8040405 123.195959,58 100,58 Z"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
      <div
        class="styledComponents__BlockContainer-sc-1b51i62-0 fYurXy"
      >
        <h3
          class="styledComponents__Heading-sc-1b51i62-2 frvBdD"
        >
          Effective Escrow Breakdown
        </h3>
        <div
          class="styledComponents__ChartContainer-sc-1b51i62-1 keWEnA"
        >
          <div
            class="styledComponents__SkeletonContainer-sc-1b51i62-4 jVKzvb"
          >
            <svg
              height="0"
              style="position: absolute;"
              width="0"
            >
              <defs>
                <clippath
                  id="donutClipPath"
                >
                  <path
                    d="M100,0 C155.228475,0 200,44.771525 200,100 C200,155.228475 155.228475,200 100,200 C44.771525,200 0,155.228475 0,100 C0,44.771525 44.771525,0 100,0 Z M100,58 C76.8040405,58 58,76.8040405 58,100 C58,123.195959 76.8040405,142 100,142 C123.195959,142 142,123.195959 142,100 C142,76.8040405 123.195959,58 100,58 Z"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="GradientBorder-zuziiu-0 cgqNXZ"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1wpt79b-0 cUQAml"
  >
    <div
      class="styledComponents__TopStripe-sc-1wpt79b-4 fawtJw"
      color="rgb(255, 191, 0)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1wpt79b-2 fLRQBK"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 1.8rem; width: 1.8rem;"
        viewBox="0 0 64 64"
      >
        <circle
          cx="32"
          cy="31.964"
          fill="rgb(80, 103, 55)"
          r="21"
        />
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1wpt79b-1 ikQNuj"
      >
        Taxes
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1wpt79b-3 eVQUHf"
      />
    </div>
    <table
      class="styledComponents__StyledTable-sc-1gw5eqk-0 styledComponents__CustomTable-sc-34ple3-0 iysqfc"
    >
      <thead
        class="styledComponents__StyledTableHead-sc-1gw5eqk-3 ezRwGt"
      >
        <tr
          class="styledComponents__StyledTableRow-sc-1gw5eqk-5 styledComponents__TableHeadRow-sc-34ple3-1 hTMbUf"
        >
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Tax Type
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Paid To
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Tax ID
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Next Due
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Expected Amount
          </th>
        </tr>
      </thead>
      <tbody
        class="styledComponents__StyledTableBody-sc-1gw5eqk-1 iLjHxu"
      >
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
      </tbody>
    </table>
    <div
      class="GradientBorder-zuziiu-0 cgqNXZ"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1wpt79b-0 cUQAml"
  >
    <div
      class="styledComponents__TopStripe-sc-1wpt79b-4 fawtJw"
      color="rgb(255, 191, 0)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1wpt79b-2 fLRQBK"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 1.8rem; width: 1.8rem;"
        viewBox="0 0 64 64"
      >
        <circle
          cx="32"
          cy="31.964"
          fill="rgb(80, 103, 55)"
          r="21"
        />
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1wpt79b-1 ikQNuj"
      >
        Homeowner's/Hazard Insurance
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1wpt79b-3 eVQUHf"
      />
    </div>
    <table
      class="styledComponents__StyledTable-sc-1gw5eqk-0 styledComponents__CustomTable-sc-5wfs9q-0 jcQqKD"
    >
      <thead
        class="styledComponents__StyledTableHead-sc-1gw5eqk-3 ezRwGt"
      >
        <tr
          class="styledComponents__StyledTableRow-sc-1gw5eqk-5 styledComponents__TableHeadRow-sc-5wfs9q-1 jmAleU"
        >
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Insurance Type
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Policy Type
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Company
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Policy Expiration
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Annual Premium
          </th>
        </tr>
      </thead>
      <tbody
        class="styledComponents__StyledTableBody-sc-1gw5eqk-1 iLjHxu"
      >
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
      </tbody>
    </table>
    <div
      class="GradientBorder-zuziiu-0 cgqNXZ"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1wpt79b-0 cUQAml"
  >
    <div
      class="styledComponents__TopStripe-sc-1wpt79b-4 fawtJw"
      color="rgb(255, 191, 0)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1wpt79b-2 fLRQBK"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 1.8rem; width: 1.8rem;"
        viewBox="0 0 64 64"
      >
        <circle
          cx="32"
          cy="31.964"
          fill="rgb(80, 103, 55)"
          r="21"
        />
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1wpt79b-1 ikQNuj"
      >
        Mortgage Insurance
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1wpt79b-3 eVQUHf"
      />
    </div>
    <table
      class="styledComponents__StyledTable-sc-1gw5eqk-0 styledComponents__CustomTable-u9rcp7-0 jIuhyg"
    >
      <thead
        class="styledComponents__StyledTableHead-sc-1gw5eqk-3 ezRwGt"
      >
        <tr
          class="styledComponents__StyledTableRow-sc-1gw5eqk-5 styledComponents__TableHeadRow-u9rcp7-1 kQlluT"
        >
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Company
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Insurance ID
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Premium Amount
          </th>
          <th
            class="styledComponents__StyledTableHeader-sc-1gw5eqk-4 kYPULa"
          >
            Premium Due
          </th>
        </tr>
      </thead>
      <tbody
        class="styledComponents__StyledTableBody-sc-1gw5eqk-1 iLjHxu"
      >
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
        <tr
          class="styledComponents__TableRowContainer-m3am64-2 cKKBAR"
        >
          <td
            class="styledComponents__TableRowStripe-m3am64-3 hQAnzD"
          />
        </tr>
      </tbody>
    </table>
    <div
      class="GradientBorder-zuziiu-0 cgqNXZ"
    />
  </section>
</section>
`;
